pkg/utils/watchJob.go

```diff
@@ -83,3 +86,25 @@ func JobLogs(experimentName string, engineName string, client *kubernetes.Client
 	}
 	return 0, nil
 }
+
+//ResultUpdate will update the result of pipelines in a table by calling a python script result_update.py
+func ResultUpdate(experimentName string, engineName string, clientSet *chaosClient.LitmuschaosV1alpha1Client) (int, error) {
+
+	//Updating the result table
+	dt := time.Now()
+	app, err := clientSet.ChaosResults(chaosTypes.ChaosNamespace).Get(engineName+"-"+experimentName, metav1.GetOptions{})
+	Expect(err).To(BeNil(), "Fail to get the chaosresult while updating the result in a table")
+	testVerdict := string(app.Status.ExperimentStatus.Verdict)
+	fmt.Println("The job_id for the job will be", os.Getenv("CI_JOB_ID"))
+	fmt.Println("The testVerdict for the experiment will be", testVerdict)
+	cmd := exec.Command("python3", "../utils/result_update.py", "--job_id", os.Getenv("CI_JOB_ID"), "--stage", "Experiment", "--test_desc", experimentName, "--test_result", testVerdict, "--time_stamp", (dt.Format(time.ANSIC)), "--token", os.Getenv("GITHUB_TOKEN"), "--test_name", experimentName)
+	out, err := cmd.Output()
+
+	if err != nil {
+		println(err.Error())
+		return 1, err
+	}
+	fmt.Println(string(out))
```

Comment:
Could you please post these logs ?

Reply:
Modified it more to handle both errors Gitlab error along with python errors:
This is the output for a successful run:
```
------------------------------
The job_id for the job is 61254
The testVerdict for the experiment is Pass
Trying to update readme.md file at path: tools/README.md
README.md content update try: 5
Readme.md updated successfully

Result updated successfully !!!
•
Ran 2 of 2 Specs in 64.754 seconds
SUCCESS! -- 2 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestChaos (64.75s)
```

This is the output of fail case (suppose I failed it giving wrong branch name):

```
The job_id for the job will be 
The testVerdict for the experiment will be Pass
exit status 1: Traceback (most recent call last):
  File "../utils/result_update.py", line 39, in <module>
    contents = repo.get_contents("tools/README.md", "basel32142ine")
  File "/home/gitlab-runner/.local/lib/python3.5/site-packages/github/Repository.py", line 1706, in get_contents
    parameters=url_parameters,
  File "/home/gitlab-runner/.local/lib/python3.5/site-packages/github/Requester.py", line 319, in requestJsonAndCheck
    verb, url, parameters, headers, input, self.__customConnection(url)
  File "/home/gitlab-runner/.local/lib/python3.5/site-packages/github/Requester.py", line 342, in __check
    raise self.__createException(status, responseHeaders, output)
github.GithubException.GithubException: 404 {"message": "No commit found for the ref basel32142ine", "documentation_url": "https://developer.github.com/v3/repos/contents/"}
```